mobify/spline
# Prepare Spline for NPM

Reviewers: @marlowpayne, @mikenikles 
CC: @yourpalsonja @cole-sanderson @mlworks @Magitron

## Changes
- Remove Bower as the distribution network
- Prepare this repo for distribution on NPM
- Prepare a new version of Spline with new features
    - New utility mixins
    - Removed some irrelevant mixins

## Todos:
- [x] Add missing utility mixins, which are to be used in Progressive Web SDK and projects (scaffold)
- [x] +1 Engineer (@marlowpayne)
- [x] +1 UI Dev (@cole-sanderson)
- [ ] Add tests for new mixins (maybe a post merge change)

## How to Test
- Testing the Repo itself
    - Follow the "How To Test" instructions in the readme
- Test in [SDK](https://github.com/mobify/progressive-web-sdk)
    - Clone the SDK repo with `git clone https://github.com/mobify/progressive-web-sdk.git`
    - To the SDK's `package.json` add this as a `devDependencies`:

        ```
        "spline": "git@github.com:mobify/spline.git#prepare-for-npm"
        ```

    - In the cloned repo, run `npm install`
    - In one of the SDK components, add `@import "node_modules/spline-scss/dist/spline";`
    - Apply one of Spline's mixins to the component SCSS
    - Run `npm run styleguide`
    - Open [http://0.0.0.0:4000](http://0.0.0.0:4000) and find the component you tweaked
    - Confirm that the mixin styles were applied
- Test in [Scaffold](https://github.com/mobify/progressive-web-scaffold)
    - Clone the scaffold repo with `git clone https://github.com/mobify/progressive-web-scaffold.git`
    - To the Scaffold's `package.json` add this as a `devDependencies`:

        ```
        "spline": "git@github.com:mobify/spline.git#prepare-for-npm"
        ```

    - In the cloned repo, run `npm install`
    - Add `@import "node_modules/spline-scss/dist/spline";` into `/app/stylesheet.scss`
    - Apply one of Spline's mixins to any class in in one of the `/app/styles/**/*.scss` files
    - Add that class to an element in an existing page/container or component in the scaffold
    - Run `npm run dev`
    - Open [Mobify Preview](https://preview.mobify.com/?url=http%3A%2F%2Fwww.merlinspotions.com%2F&site_folder=https%3A%2F%2Flocalhost%3A8443%2Floader.js&disabled=0&domain=&scope=1) and click "Preview"
    - Confirm that the mixin styles were applied
